setup_hooks.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+rustup component add rustfmt-preview
+
+rustfmt_path=`find $HOME/.rustup/toolchains/ -wholename '*bin/rustfmt*'`
+echo "#!/bin/sh
+find . -name *.rs -type f -print | xargs $rustfmt_path"  > .git/hooks/pre-commit
```

Comment:
This only finds *.rs in the top leve.

Also `xargs` will just write all the diffs, and the return code is 0, i.e. the commit will pass.
```
$ git add src/lib.rs
$ git commit -m "small tweaks" // succeed despite format issue
$ git status
Changes not staged for commit:
   modified: src/lib.rs
```

We would want the commit to fail if the format is wrong.

Reply:
This is pre-commit hook, it have nothing with passing build.

The idea is that guy has broken formatting, it writes `git commit` and hook just fixes all issues for him. It's better to fix an issue automatically than blame user and ask him to fix all manually.